magefiles/build.go

```diff
@@ -0,0 +1,27 @@
+//go:build mage
+
+//nolint:wrapcheck
+package main
+
+import "github.com/magefile/mage/sh"
+
+// Build a Python wheel.
+func Build() error {
+	if err := sh.RunV("python3", "-mvenv", "foo"); err != nil {
+		return err
+	}
+
+	if err := sh.RunV("sh", "-c", "'source ./env/bin/activate'"); err != nil {
```

Comment:
This doesn't work and I'm not sure what to do with this because this is programmatically.

Reply:
this is because this command only activates the environment for _this_ shell, and thus then next `pip install` invocation won't use it

you can change remove it and change all the further invocation to be prefixed by the venv location, e.g. `./env/bin/pip install`, `./env/bin/python -mziglang cc`, etc.